### DISCUSSION TITLE
About Olive and ONNX

### DISCUSSION THREAD

**Discussion by Helius22** [lshqqytiger/stable-diffusion-webui-directml#127] [Q&A]
So from what I understood ONNX and Olive can be used to optimize models, does this webui have support for olive / onnx models? If no will it be added? Where can I convert my models is there another GUI for that? 

This seems like a big deal for AMD users but I've seen barely any talk about it, other than all the articles about "2x nvidia performance in new driver".

Oh and does anyone have information about AMD supporting these optimizations for the 6000-series? In [microsoft's blog](https://devblogs.microsoft.com/directx/dml-stable-diffusion/) it says  _"AMD plans to release optimized graphics drivers in the next month supporting AMD RDNA™ 3 devices including AMD Radeon™ RX 7900 Series graphics"_ but how does that make sense if olive optimizes the model and not SD itself? Shouldn't it run on any card capable of DX12? 

-I'm confused

**Comment by lshqqytiger** [comment]
AMDGPUs support Olive (because they support DX12).
But, at that moment, webui is using PyTorch only, not ONNX.
So, in order to add Olive optimization support to webui, we should change many things from current webui and it will be very hard work.
You can find the model optimizer (which automatically convert your models but they must be ONNX models, not .ckpt/.safetensors) and very simple GUI from https://github.com/microsoft/Olive/tree/main/examples/directml/stable_diffusion
